### ISSUE TITLE
Spurious interning error in for-each loop: "found `long`, required `@Interned long`"

### ISSUE THREAD

**Issue by Calvin-L** [typetools/checker-framework#6371]
```java
    public void interningLongs(List<Long> list) {
        for (long i : list) {
            long unused = Math.addExact(i, 0L);
        }
    }
```

For the above code, the Interning Checker reports

```
error: [argument] incompatible argument for parameter x of Math.addExact.
   found   : long
   required: @Interned long
```

It is true that the boxed capital-L `Long` values in the list are not interned, but I believe they should be treated as interned once they are unboxed to primitive lower-case-L `long` values.
